### PR TITLE
Add check for node readiness when recovering workers

### PR DESCRIPTION
**Pull request by enxebre**
`e2e-aws-operator` suite for machine API is aggressively being hit by a networking issue when scaling machines, some new nodes do not go ready - https://bugzilla.redhat.com/show_bug.cgi?id=1698624 
This ensure we fail earlier if a regression might be introduced

### CODE REVIEW

**Comment by enxebre** [comment]
cc @derekwaynecarr @smarterclayton @ingvagabund

File: test/extended/machines/workers.go [hunk 1]
```diff
@@ -106,6 +106,15 @@ func mapMachineNameToNodeName(machines []objx.Map, nodes []corev1.Node) (map[str
 	return result, len(machines) == len(result)
 }
 
+func IsNodeReady(node corev1.Node) bool {
```
Comment (vikaschoudhary16) [inline test/extended/machines/workers.go]: Should this func be global?

Comment (enxebre) [reply test/extended/machines/workers.go]: made it local, thanks!

**Comment by vikaschoudhary16** [comment]
Should this TODO be removed as well, https://github.com/openshift/origin/pull/22540/files#diff-adae07b320711a5893778c769390573eR223 ?

**Comment by squeed** [comment]
Will this cause flakes? Nodes can go NotReady if they've rebooted and the SDN hasn't come back up. It should be relatively quick between boot and Ready, but the window is definitely there.

**Comment by ingvagabund** [comment]
> Will this cause flakes? Nodes can go NotReady if they've rebooted and the SDN hasn't come back up. It should be relatively quick between boot and Ready, but the window is definitely there.

If that's the case we just might need to wait longer. What's the likelihood of one node being in NotReady state of N nodes?

**Comment by enxebre** [comment]
@squeed This is not timing issue, I'm consistently seeing goes remaining unready for ever on my openshift-dev cluster

**Comment by squeed** [comment]
@enxebre for sure - it's probably caused by https://bugzilla.redhat.com/show_bug.cgi?id=1669096, a bad crio bug. Once that's dealt with, then we should never have any bugs ever again :-)

**Pull request merged**